clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -55,19 +72,56 @@ public JmxReporter() {
      * Create a JMX reporter that prefixes all metrics with the given string.
      */
     public JmxReporter(String prefix) {
+        this(prefix, "");
+    }
+
+    public JmxReporter(String prefix, String configPrefix) {
         this.prefix = prefix;
+        this.configPrefix = configPrefix;
     }
 
     @Override
-    public void configure(Map<String, ?> configs) {}
+    public void configure(Map<String, ?> configs) {
+        reconfigure(configs);
+    }
+
+    @Override
+    public Set<String> reconfigurableConfigs() {
+        return Utils.mkSet(configPrefix + JmxReporter.WHITELIST_CONFIG,
+                           configPrefix + JmxReporter.BLACKLIST_CONFIG);
+    }
+
+    @Override
+    public void validateReconfiguration(Map<String, ?> configs) throws ConfigException {
+        compilePredicate(configPrefix, configs);
+    }
+
+    @Override
+    public void reconfigure(Map<String, ?> configs) {
+        synchronized (LOCK) {
+            this.mbeanPredicate = JmxReporter.compilePredicate(configPrefix, configs);
```

Comment:
I don't see a good reason to make this a `java.util.function.Predicate` rather than just a regular instance method that we call.  We're always calling the same function (the predicate function is never something else).

I agree that we should compile the whitelist and blacklist regex ahead of time, but that seems more straightforward to just do in the `reconfigure` function itself.

Reply:
having a predicate simplifies reusing it in both Yammer and Kafka metrics.

Reply:
I might be missing something, but in both cases, you just want to use regular expressions, right?  There is no need to mess around with predicate functions.

Reply:
It's not clear to me what you are suggesting to replace it with.

I am calling compilePredicate method in two different classes:

https://github.com/apache/kafka/pull/7674/files#diff-717683fa5572a1685d3dfaadeef5c094R76
and 
https://github.com/apache/kafka/pull/7674/files#diff-3acf682bd1d689a6ee23fe641d57e72aR102

I felt it was cleaner to use a predicate than to duplicate the code compiling the regexes and having each class have its own method returning `match(regexA) and !match(regexB)` to keep the logic consistent.

Reply:
On second thought, I'm fine with keeping the predicate.